flashinfer/prefill.py

```diff
@@ -2173,6 +2167,14 @@ def run(
             if self._jit_module is not None:
                 run_args.extend(list(args))
             else:
+                # Extract FP8 scale tensors from *args if q is FP8
+                fp8_scale_q = None
+                fp8_scale_k = None
+                fp8_scale_v = None
+                if is_float8(q) and len(args) >= 3:
```

Comment:
I am not a fan of making these are implicit positional args. Instead, I think we should document it clearly and make them explicit.

For example, we can make `k_scale : Optional[torch.Tensor, float]` or have separate keyword args like `k_scale_device`. Also, we should document clearly that these scale tensors should be per-head. Otherwise, illegal mem access will occur.